File: opcua-webservice-spring/core/src/main/java/org/opcfoundation/webapi/service/types/ServiceContext.java
package org.opcfoundation.webapi.service.types;

import org.eclipse.milo.opcua.stack.core.types.structured.RequestHeader;
import org.springframework.lang.Nullable;
import org.springframework.web.context.request.NativeWebRequest;

public class ServiceContext {
    final private @Nullable String serverUri;
    final private @Nullable RequestHeader requestHeader;
    final private @Nullable NativeWebRequest webRequest;

    public ServiceContext(
            @Nullable String serverUri,
            @Nullable RequestHeader header,
            @Nullable NativeWebRequest webRequest)
    {
        this.serverUri = serverUri;
        this.requestHeader = header;
        this.webRequest = webRequest;
    }

    public @Nullable String getServerUri() {
        return serverUri;
    }

    public @Nullable RequestHeader getRequestHeader() {
        return requestHeader;
    }

    public @Nullable NativeWebRequest getWebRequest() {
        return webRequest;
    }
}

File: opcua-webservice-node/src/server/types/contexts/ServiceContext.ts
import { RequestHeader } from "opcua-webapi";

export abstract class ServiceContext
{
    private _serverUri : string | null;
    private _requestHeader : RequestHeader | null;

    constructor(
        serverUri : string | undefined,
        requestHeader: RequestHeader | undefined)
    {
        this._serverUri = (serverUri) ? serverUri : null;
        this._requestHeader = (requestHeader) ? requestHeader : null;
    }

    get serverUri() : string | null
    {
        return this._serverUri;
    }

    get requestHeader() : RequestHeader | null
    {
        return this._requestHeader;
    }
}